engine/simulationcraft.hpp

```diff
@@ -6360,12 +6363,19 @@ struct real_ppm_t
     if ( last_trigger_attempt == player -> sim -> current_time() )
       return false;
 
-    bool success = player -> rng().roll( proc_chance( player, rppm, last_trigger_attempt, last_successful_trigger, scales_with ) );
-
-    last_trigger_attempt = player -> sim -> current_time();
+    double chance = proc_chance( player, rppm, last_trigger_attempt,
+                                 last_successful_trigger, scales_with );
+    bool success = player -> rng().roll( chance );
 
     if ( success )
+    {
+      first_proc_occurred = true;
       last_successful_trigger = player -> sim -> current_time();
+    }
+    // Due to a bug, last attempt to trigger is fixed until first proc
```

Comment:
Is this something that's likely to be fixed in a patch, ever? If we codify in - game bugs, is there mechanism in place to remove our implementation of the bug when the real bug is fixed?